src/libraries/Microsoft.PowerFx.LanguageServerProtocol/Protocol/CustomNL2FxParams.cs

```diff
@@ -21,6 +21,11 @@ public class CustomNL2FxParams : BaseNLParams, IHasTextDocument
         /// Sentence in Natural Language.
         /// </summary>
         public string Sentence { get; set; }
+
+        /// <summary>
+        /// Name of the NL2Fx feature being used.
+        /// </summary>
+        public string FeatureName { get; set; }
```

Comment:
Only this change is needed here

Reply:
once you complete this PR, you consume the Power fx version with your changes in pa client and then in you PA client Language server NLhandler PreHandleNl2Fx() you should be able to access FeatureName provided you also pass FeatureName  from Power fx formula bar PPUx package when nl2fx request is sent